src/tracker/tracker.py

```diff
@@ -0,0 +1,429 @@
+        self.colormap = cv2.applyColorMap(
+            np.arange(256, dtype=np.uint8).reshape(1, -1), cv2.COLORMAP_COOL
+        )[0]
+        self.count = 0
+
+    async def stop(self):
+        """
+        Stop the background loop by setting the stop event.
+        """
+        self.stop_event.set()
+        await self.background_task  # Wait for the background task to finish
+
+    async def _background_update_loop(self):
+        """
+        Background loop that continuously gets images from the camera and updates tracks.
+        """
+        while not self.stop_event.is_set():
+            viam_img = await self.camera.get_image(mime_type=CameraMimeType.JPEG)
+            img = decode_image(viam_img)
+            self.update(img)  # Update tracks
+            await self.write_detections()  # Write detections to the shared object
+            await sleep(self.sleep_period)
+
+    async def write_detections(self):
+        """
+        Write the current detections from tracks into the shared CurrentDetections object.
+        """
+        cur_tracks = {
+            track_id: self.tracks[track_id] for track_id in self.current_tracks_id
+        }
+        detections = [track.get_detection() for track in cur_tracks.values()]
+        await self.current_detections.set_detections(detections)
+
+    async def get_current_detections(self):
+        """
+        Get the current detections.
+        """
+        return await self.current_detections.get_detections()
+
+    async def is_new_object_detected(self):
+        return self.new_object_event.is_set()
+
+    def update(self, img, visualize: bool = False):
+        """
+        Update the tracker with new detections.
+
+        :param detections: List of Detection objects detected in the current frame.
+        """
+
+        # Get new detections
+        detections = self.detector.detect(img)
+
+        if not detections:
+            self.current_tracks_id = set()
+            for track in self.tracks.values():
+                track.increment_age()
+            return
+
+        # Compute feature vectors for the current detections
+        feature_vectors = self.encoder.compute_features(img, detections)
+
+        # Initialize cost matrix
+        cost_matrix = np.zeros((len(self.tracks), len(detections)))
+
+        # Create a list to keep track of updated track IDs
+        updated_tracks_ids = []
+
+        # New tracks
+        new_tracks_ids = []
+
+        # Calculate cost for each pair of track and detection
+        track_ids = list(self.tracks.keys())
+        for i, track_id in enumerate(track_ids):
+            track = self.tracks[track_id]
+            for j, detection in enumerate(detections):
+                iou_score = track.iou(detection.bbox)
+                feature_dist = euclidean(track.feature_vector, feature_vectors[j])
+
+                # Cost function: lambda * feature distance + (1 - lambda) * (1 - IoU)
+                # TODO: figure out this 30
```

Comment:
This is basically a normalization. The embeddings outputted by the encoder are not normalized, so I found this value empirically (see the Alex/Robin experimentations). I plan to test it out better once the tracker is running in a "top-view" camera